src/main.rs

```diff
@@ -256,7 +326,11 @@ impl Vec3 {
     }
 
     fn rotate(self, rot: Quat) -> Vec3 {
-        Vec3::from(rot * self * rot.conj())
+        if rot == Quat::default() {
+            self
+        } else {
+            Vec3::from(rot * self * rot.conj())
+        }
```

Comment:
`Quat::default()` is zero, which does *not* represent zero rotation. Zero rotation is represented by `Quat::one()`.

Reply:
oh! ok, good catch